cmd/relay/handlers.go

```diff
@@ -54,7 +54,14 @@ func (s *Service) handleComAtprotoSyncRequestCrawl(c echo.Context, body *comatpr
 		hostURL = "http://" + hostname
 	}
 
-	if err := s.relay.HostChecker.CheckHost(ctx, hostURL); err != nil {
+	if !s.config.AllowInsecureHosts {
+		err = s.relay.HostChecker.CheckHost(ctx, hostURL)
+	} else {
+		hc := relay.NewHostClient("")
```

Comment:
It sounds simpler to set `hc.Client = http.DefaultClient` in `NewRelay()` if `config.AllowInsecureHosts` is set.

Reply:
According to the code intelligence, `relay.HostChecker` is literally only used on that line, do you know of other users?

Reply:
Oh, I see. Yeah, with the additional check on `admin` it makes sense :+1:

Reply:
But maybe use `NewHostClient(s.config.UserAgent)` like in `NewRelay()`?

Reply:
This PR starts using some of the *account* checking behavior: https://github.com/bluesky-social/indigo/pull/1215. I guess that will probably not work with localhost stuff now that I think of it.

This is honestly a big messy tangle, I kind of want to just be secure all the time and not support the insecure/localhost mode at all.